crates/rpc/rpc/src/eth/filter.rs

```diff
@@ -417,7 +418,7 @@ where
         trace!(target: "rpc::eth::filter", from=from_block, to=to_block, ?filter, "finding logs in range");
```

Comment:
we can simplify this a bit more:

this function is always called after fetching the `ChainInfo`:

https://github.com/paradigmxyz/reth/blob/main/crates/rpc/rpc/src/eth/filter.rs#L372-L372

https://github.com/paradigmxyz/reth/blob/main/crates/rpc/rpc/src/eth/filter.rs#L134-L134

so we can use this as an argument to this function

then check if `to_block == from_block == info.best_number`